actionpack/lib/action_dispatch/routing/route_set.rb

```diff
@@ -363,7 +363,7 @@ def self.default_resources_path_names
       end
 
       def self.new_with_config(config)
-        route_set_config = DEFAULT_CONFIG
+        route_set_config = DEFAULT_CONFIG.dup
```

Comment:
This has to be a bug. We mutate a constant in the next few lines which changes the config of future route sets. Duping the constant prevents this.